src/renderer/components/dialog-add-version.tsx

```diff
@@ -103,7 +112,18 @@ export class AddVersionDialog extends React.Component<
       name,
     };
 
-    this.props.appState.addLocalVersion(toAdd);
+    // swap to old local electron version if the user adds a new one with the same path
+    const shouldSetToPreviousVersion = isValidElectron && existingLocalVersion;
+    if (shouldSetToPreviousVersion) {
+      // set previous version as active version
+      for (const ver of Object.values(versions)) {
+        if (ver.localPath === existingLocalVersion?.localPath) {
+          this.props.appState.setVersion(ver.version);
+        }
+      }
```

Comment:
Could this for loop be replaced with `this.props.appState.setVersion(existingLocalVersion.localPath)`?

It looks like there's something going on here that makes walking the `appState.versions` array necessary, but I'm not seeing why that's needed instead of using `existingLocalVersion.localPath` :thinking:

Reply:
Oops! You are totally right- an oversight for when line 121 used to delete the previous version in a previous commit that I ended up changing 😆